src/main/java/com/minecolonies/coremod/tileentities/TileEntityRack.java

```diff
@@ -443,6 +482,8 @@ else if (compound.hasKey(TAG_RELATIVE_NEIGHBOR))
         }
         main = compound.getBoolean(TAG_MAIN);
         updateItemStorage();
+
+        this.inWarehouse = compound.getBoolean(TAG_IN_WAREHOUSE);
```

Comment:
I like to put a "hasCompoundKey" wrapper around it, to not rely on the default value

Reply:
it returns false if there is no key. (just looked at the code)

Reply:
(which, if it's false, that's correct)